pkg/netpol/eval/ingressanalyzer/ingress_analyzer.go

```diff
@@ -156,28 +140,44 @@ func (ia *IngressAnalyzer) getServiceFromServiceNameAndNamespace(svcName, svcNam
 ////////////////////////////////////////////////////////////////////////////////////
 // Ingress connections
 
-// AllowedIngressConnectionsToAWorkloadPeer return the allowed external ingress-controller's connections to the dst peer
-func (ia *IngressAnalyzer) AllowedIngressConnectionsToAWorkloadPeer(dst eval.Peer) (eval.Connection, error) {
-	// if there is at least one route/ ingress object that targets a service which selects the dst peer,
-	// then we have an ingress conns to the peer
-
-	// assuming dstPeer is WorkloadPeer, should be converted to k8s.Peer
-	dstPodPeer, err := ia.pe.ConvertWorkloadPeerToPodPeer(dst)
-	if err != nil {
-		return nil, err
+func (ia *IngressAnalyzer) getRouteTargetedPods(namespace string, svcList []string) []*k8s.Pod {
+	var res []*k8s.Pod
+	for _, svc := range svcList {
+		svcPods, err := ia.getServicePods(svc, namespace)
+		if err != nil {
+			continue // can not get targeted pods of this service
+		}
+		res = append(res, svcPods...)
 	}
+	return res
+}
 
-	peerNs := dstPodPeer.Namespace()
-	rtMap, ok := ia.routesMap[peerNs]
-	if !ok {
-		return nil, nil // no ingress objects in the pod's namespace
-	}
+// AllowedIngressConnections returns map of the allowed external ingress-controller's connections of each targeted peer
+func (ia *IngressAnalyzer) AllowedIngressConnections() map[*k8s.Pod]eval.Connection {
```

Comment:
The exported function returns a map that its key is of type which is internal.. also, is it safe that the map key is pointer?
should consider using the exported type eval.Peer ? or name string (in level of workload or pod)?